src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_INITIAL_CAPACITY;
+        this.table = (Node<K, V>[]) new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= capacity * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
 
+        Node<K, V> newNode = new Node<>(key == null ? 0 : key.hashCode(), key, value, table[index]);
```

Comment:
The handling of null keys in the `put` method is not entirely correct. While you are assigning a hash of 0 for null keys, you should also ensure that the `getIndex` method can handle null keys consistently. Consider using a separate bucket for null keys to avoid collisions with other keys.